### PR TITLE
fixes #24323 - add post processing framework

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

File: app/models/host/discovered.rb [hunk 1]
```diff
@@ -286,6 +223,11 @@ def notification_recipients_ids
     (org_recipients + admins).uniq
   end
 
+  def render_lock_template(template)
```
Comment (timogoebel) [inline app/models/host/discovered.rb]: The old renderer needs the templates to be rendered inside the host object. We can remove this with the new renderer.

Comment (lzap) [reply app/models/host/discovered.rb]: Note for myself: if we add dependency on the new renderer I must merge this after 12.1 or 13 is out.

File: app/services/foreman_discovery/import_hooks/lldp_neighbor.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Automatically bonds interfaces based on lldp facts
+module ForemanDiscovery
+  module ImportHooks
+    class LldpNeighbor < ImportHook
+      def after_populate
+        return unless Setting[:discovery_auto_bond]
+        ::ForemanDiscovery::LldpNeighbors.eventually_make_bond(host)
```
Comment (lzap) [inline app/services/foreman_discovery/import_hooks/lldp_neighbor.rb:7]: This could be moved into this class perhaps.

**Comment by lzap** [comment]
Nice thanks a bunch!

**Comment by lzap** [comment]
Merge conflicts, I am going to fix tests first, release v13 and then

**Comment by timogoebel** [comment]
Rebased.

**Comment by lzap** [comment]
Tests are failing in develop working on it thanks! Need to get it green first.

File: app/services/foreman_discovery/import_hooks/lldp_neighbor.rb (hunk 2 above)
Comment (lzap) [reply app/services/foreman_discovery/import_hooks/lldp_neighbor.rb:7]: I believe this definitely belongs here. What I mean is kill that LLDP helpers and move everything into this import hook.

**Comment by lzap** [comment]
Looks good, I am assuming you made no changes to the code itself, if you did please don't. This is supposed to be just refactoring without any changes. Thanks!

Discovery develop is finally green again, please rebase to retest on jenkins.

File: app/services/foreman_discovery/import_hooks/lldp_neighbor.rb (hunk 2 above)
Comment (timogoebel) [reply app/services/foreman_discovery/import_hooks/lldp_neighbor.rb:7]: Sure, we can move this in a separate PR.

**Comment by timogoebel** [comment]
Rebased.

File: app/services/foreman_discovery/import_hooks/lldp_neighbor.rb (hunk 2 above)
Comment (lzap) [reply app/services/foreman_discovery/import_hooks/lldp_neighbor.rb:7]: Timo 1 me 0. Okay :-)

**Comment by lzap** [comment]
All failed, syntax error or something perhaps.

File: app/services/foreman_discovery/import_hooks/lldp_neighbor.rb (hunk 2 above)
Comment (timogoebel) [reply app/services/foreman_discovery/import_hooks/lldp_neighbor.rb:7]: I simply don't have time to look at this now. So let's take what we already have for now. Small steps :-P

**Comment by timogoebel** [comment]
Fixed the syntax error.

**Comment by lzap** [comment]
Ok this is good but five tests are still failing: `NoMethodError: undefined method `render' for "test":String` for lock templates.

File: app/services/foreman_discovery/import_hooks/lock_templates.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    class LockTemplates < ImportHook
+      def after_populate
+        lock_templates if Setting::Discovered.discovery_lock? && host.subnet.tftp?
+      end
+
+      def lock_templates
+        TemplateKind::PXE.each do |kind|
+          template_name = Setting["discovery_#{kind.downcase}_lock_template"]
+          logger.info "Locking discovered host #{host.mac} in subnet #{host.subnet} via #{template_name} template"
+          template = ::ProvisioningTemplate.find_by_name(template_name).template
```
Comment (lzap) [inline app/services/foreman_discovery/import_hooks/lock_templates.rb]: Here is it, remove `.template` and that should do it.

**Comment by timogoebel** [comment]
@lzap: Removed `.template` as you proposed, thanks!

**Comment by lzap** [comment]
Hmmm Test Result (3 failures / -2)
AuditMailerTest.test_0006_Audit template change should not crash
SeedsTest.test_0006_doesn't add a template back that was renamed
SeedsTest.test_0005_doesn't add a template back that was deleted

[test]

**Comment by timogoebel** [comment]
[test foreman_discovery]

**Review by lzap — APPROVED** [review]
Thanks

**Pull request merged**